### PR TITLE
[Alignment configuration] Free end gap configuration - documentation and removal of aligned_ends

### PR DESCRIPTION
**Pull request by smehringer**
Final PR of #2032
Closes #2032

Resolves seqan/product_backlog/issues/96

### CODE REVIEW

**Comment by marehr** [comment]
Can you rebase and make it compile?

**Review by marehr — CHANGES_REQUESTED** [review]
(Can you make it compile?)

**Comment by smehringer** [comment]
Oh, when I made the PR it at least was mergeable.
Sure

**Comment by smehringer** [comment]
@marehr everything should be fine now

**Comment by smehringer** [comment]
Ok now even the snippets should be fine.

File: doc/tutorial/pairwise_alignment/configurations.cpp [hunk 1]
```diff
@@ -33,18 +33,16 @@
...
-seqan3::back_end_second bes{false};
-
-auto cfg_1 = seqan3::align_cfg::aligned_ends{seqan3::end_gaps{fef, bef, fes, bes}};
-auto cfg_2 = seqan3::align_cfg::aligned_ends{seqan3::end_gaps{fef, fes}};
-//! [aligned_ends]
-(void) cfg_1;
-(void) cfg_2;
+//! [method_global_free_end_gaps]
+    // Example of a semi-global alignment where leading and trailing gaps in the
+    // second sequence are not penelised:
```
Comment (marehr) [inline doc/tutorial/pairwise_alignment/configurations.cpp:38]:
```suggestion
    // Example of a semi-global alignment where leading and trailing gaps in the
    // second sequence are not penalised:
```

File: doc/tutorial/pairwise_alignment/index.md [hunk 2]
```diff
@@ -306,20 +287,27 @@ To make the configuration easier, we added a shortcut called seqan3::align_cfg::
...
+can be further refined with free end gaps (see section `Global and semi-global alignment`).
+
+\attention Only the following free end gap configurations are supported for the
+global alignment configuration with the edit scheme:
+- no free end gaps (all free end gap specifiers are set to `false`)
+- free end gaps for the first sequence (free end gaps are set to `true` for the first and
+  to `false` for the second sequence)
+Using any other free end gap configuration will
+disable the edit distance and fall back to the standard pairwise alignment and will not use the fast bitvector
+algorithm.
```
Comment (marehr) [inline doc/tutorial/pairwise_alignment/index.md:301]: reflow this text

File: doc/tutorial/pairwise_alignment/index.md [hunk 3]
```diff
@@ -306,20 +287,27 @@ To make the configuration easier, we added a shortcut called seqan3::align_cfg::
...
+
 ### Refine edit distance
 
-The edit distance can be further refined using seqan3::align_cfg::aligned_ends to also compute a semi-global alignment
-and the seqan3::align_cfg::min_score configuration to fix an edit score (a limit of the allowed number of edits). If the
+The edit distance can be further refined using the seqan3::align_cfg::min_score configuration to fix an edit score (a limit of the allowed number of edits).. If the
```
Comment (marehr) [inline doc/tutorial/pairwise_alignment/index.md:305]: line limit and reflow following text

**Review by marehr — CHANGES_REQUESTED** [review]
Just small documentation stuff; thank you! :)

**Review by marehr — APPROVED** [review]
ty :)

**Review by rrahn — APPROVED** [review]

**Comment by rrahn** [comment]
can you rebase real quick? Or can I squash merge?

**Pull request merged**